src/screens/Watch/Components/CTPlayer/player_hls.jsx

```diff
@@ -307,36 +308,45 @@ const Video = React.memo((props) => {
             } else {
                 englishTrack = textTrack[0];
             }
+            
              dispatch({type: "watch/setEnglishTrack", payload: englishTrack});
+            
 
             englishTrack.addEventListener("cuechange", (event) => {
                 
                 // console.log(event);
                 const toLog = [];
                 for (let z = 0; z < event.currentTarget.cues.length; z++) {
-                    toLog.push(event.currentTarget.cues[z])
+                    let toCopy = JSON.parse(JSON.stringify(event.currentTarget.cues[z]));
+                    toCopy.startTime = event.currentTarget.cues[z].startTime;
+                    toCopy.endTime = event.currentTarget.cues[z].endTime;
+                    toCopy.text = event.currentTarget.cues[z].text;
+                    toLog.push(Object.freeze(toCopy))
                 }
 
                 // console.log(toLog)
-                const l = event.currentTarget.activeCues[0];
+             
                 // const prev = undefined;
-                if (l !== undefined) {
+                if (event.currentTarget.activeCues[0] !== undefined) {
                     idR += 1
+                    let curr = event.currentTarget.activeCues[0];
+                    if (Math.abs(curr.startTime - curr.endTime) > 20) {
```

Comment:
This code needs comments to explain *why*
Also, why 20? Are there known upper/lower bounds to this value?

Reply:
CBS and others have a max of 10 - 15 seconds. In testing anything around 20 was 100% an illegitimate caption. 

A different way I just thought of which may be preferred... if a caption overlaps with more than one other we delete it.

Reply:
So please add something like the following to the source code
// In testing with seeking the current version of HLS (VERSION 1234) ocasionally creates erroneous cues
// these stay on the screen for a long time
// we don't want these in the transcript. Testing found these typically have a dispay time >20 seconds whereas
// bonafide cues have a caption time < 15seconds
// An alternative filter might ignore a cue that already overlaps with more than one; however what if bad cue is 'new'?